filebeat/publish/publish.go

```diff
@@ -127,8 +127,8 @@ func (p *syncLogPublisher) Start() {
 }
 
 func (p *syncLogPublisher) Stop() {
-	close(p.done)
 	p.client.Close()
+	close(p.done)
```

Comment:
not really. libbeat is not checking connections being closed yet (disabled due to races). There is still a chance events are published besides being shutdown.